addon/components/phone-input.js

```diff
@@ -182,23 +189,39 @@ export default Component.extend({
   },
 
   willDestroyElement() {
-    this._iti.destroy();
+    this._iti?.destroy();
     this.element.removeEventListener('countrychange', this._onCountryChange);
 
     this._super(...arguments);
   },
 
   async _loadAndSetup() {
-    await this.phoneInput.load();
-
-    this._setupLibrary();
-
-    this._formatNumber();
-
-    this.element.addEventListener(
-      'countrychange',
-      this._onCountryChange.bind(this)
-    );
+    try {
+      this.set('isLoadingIti', true);
+
+      await this.phoneInput.load();
+
+      // Even if the above promise resolves, it might be at the end of the
+      // component lifecycle
+      if (this.isDestroying || this.isDestroyed) {
+        return;
+      }
+
+      this._setupLibrary();
+
+      this._formatNumber();
+
+      this.element.addEventListener(
+        'countrychange',
+        this._onCountryChange.bind(this)
+      );
+    } catch (error) {
+      console.error('Failed to load intl-tel-input');
```

Comment:
Wouldn't it be better to throw rather than log the error?

Reply:
Digging deeper reveals others implemented their own solution to overcome the same issue: https://github.com/kaliber5/ember-bootstrap/pull/1581/files#diff-f1dd2645d1cfeba1b764abf0b4f904a04852ab7f77a6d8d0701830b7f880a46c